src/main/java/uk/gov/hmcts/dts/fact/mapit/MapitValidator.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.hmcts.dts.fact.mapit;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.dts.fact.services.MapitService;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Component
+public class MapitValidator {
+
+    private final MapitService mapitService;
+    private static final String FULL_POSTCODE = "([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z]"
+        + "[A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y]"
+        + "[0-9][A-Za-z]?))))\\s?[0-9][A-Za-z]{2})";
+    private static final String PARTIAL_POSTCODE_SEARCH = "[A-Za-z]{2}";
+    private static final String PARTIAL_POSTCODE_NUMERIC = "(^[A-Za-z]{1,2}([0-9]{1,3}))$";
+    private static final String PARTIAL_POSTCODE_EDGECASE = "(^[A-Za-z][0-9]{1,2}[A-Za-z][0-9]?$)|"
+        + "(^[A-Za-z]{2}([0-9]{1}[A-Za-z][0-9]?)$)";
+
+    @Autowired
+    public MapitValidator(MapitService mapitService) {
+        this.mapitService = mapitService;
+    }
+
+    public boolean postcodeDataExists(final String postcode) {
```

Comment:
We seen in the database the other day that some courts have partial postcodes with just postcode areas, like 'RM' or 'M' only. Did we say we want to just accept those postcodes as valid? Think currently they are returned as invalid?

Reply:
MAPIT returns them as invalid, so I think at the moment we are going with that for now? I think we said they would add in the full postcode in that event, or add onto the partial